### PR TITLE
[AD-681] Add all_prices method to variant decorator

### PR DESCRIPTION
**Pull request by mkamien1993**
### What problem is the code solving?
- We add this new method to return variant prices information in a way that is useful for then indexing these variants in Algolia.

### How does this change address the problem?
- Returning variant price information for each country.

### CODE REVIEW

File: app/models/spree/flow_io_variant_decorator.rb [hunk 1]
```diff
@@ -126,16 +128,64 @@ def price_in_zone(currency, product_zone)
...
+        flow_experience_key = zone&.flow_io_experience
+        next if flow_experience_key.blank?
+
+        add_flow_prices(flow_experience_key, countries, zone.flow_io_experience_country)
+      end
+
+      countries
+    end
+
+    def add_prices_for_available_currencies(zone, all_prices)
```
Comment (sebastiandl) [inline app/models/spree/flow_io_variant_decorator.rb:159]: would it make more sense for this `add_` methods to be private?

File: app/models/spree/flow_io_variant_decorator.rb [hunk 2]
```diff
@@ -126,16 +128,64 @@ def price_in_zone(currency, product_zone)
...
       all_prices
     end
 
+    def all_prices(zones)
+      countries = {}
+
+      zones.each do |zone|
+        all_prices = []
+
+        add_prices_for_available_currencies(zone, all_prices)
```
Comment (sebastiandl) [inline app/models/spree/flow_io_variant_decorator.rb:147]: is it ok for this method to modify the `all_prices` ? would it make more sense that the method returns the all_prices and that would be stored within the local all_prices? That way we could even be able to reuse this method if needed somewhere else?

**Review by sebastiandl — APPROVED** [review]

File: app/models/spree/flow_io_variant_decorator.rb (hunk 1 above)
Comment (mkamien1993) [reply app/models/spree/flow_io_variant_decorator.rb:159]: Definitely!

File: app/models/spree/flow_io_variant_decorator.rb (hunk 2 above)
Comment (mkamien1993) [reply app/models/spree/flow_io_variant_decorator.rb:147]: Good point! Changed!

**Comment by mkamien1993** [comment]
Closing this PR because these changes were added in `mejuri-web`.

**Pull request closed without merging**